### PR TITLE
Debug costmap_converter stopping

### PR DESCRIPTION
**Pull request by tonynajjar**
Not to be merged, for demonstration purposes only.

Added a monitor that checks that we are in fact publishing (not stuck). If we are stuck, some additional logs will print out for 10 seconds for further debugging, and then the node is respawned to unblocked

### CODE REVIEW

File: costmap_converter/src/costmap_converter_node.cpp [hunk 1]
```diff
@@ -125,20 +125,52 @@ class CostmapStandaloneConversion : public rclcpp::Node {
...
     pub_timer_ = n_->create_wall_timer(
         std::chrono::milliseconds(200),
-        std::bind(&CostmapStandaloneConversion::publishCallback, this));
+        std::bind(&CostmapStandaloneConversion::publishCallback, this), cb_group1_);
+    health_check_timer_ = n_->create_wall_timer(
+        std::chrono::milliseconds(5000),
+        std::bind(&CostmapStandaloneConversion::healthCheck, this), cb_group2_);
   }
 
+  void healthCheck() {
```
Comment (tonynajjar) [inline costmap_converter/src/costmap_converter_node.cpp:141]: There is a chance that we are not able to respawn and that is if the issue is that the timers die. But if that happens, that's also good input, we'll just have to accept an operations intervention

**Review by jplapp — COMMENTED** [review]
good idea. Let's run this in the basement for a while, if we don't catch it there in 1-2 hours, then we can try in production

I wonder if it makes sense to also check whether we keep receiving an updated map, or if that is equivalent to chekcing what we publish

File: costmap_converter/src/costmap_converter_node.cpp (hunk 1 above)
Comment (tonynajjar) [reply costmap_converter/src/costmap_converter_node.cpp:141]: There is also a chance that this branch fixes the issue by mistake if e.g. the issue is in the SingleThreadedExecutor and we are having deadlocks

**Comment by tonynajjar** [comment]
For https://github.com/logivations/deep_cv/pull/6159/

**Pull request closed without merging**